### DISCUSSION TITLE
How to save Pictures silently

### DISCUSSION THREAD

**Discussion by Darkhell101** [alandefreitas/matplotplusplus#233] [Q&A] [answered]
I want to save pictures silently, but every time the plot will pop out. Please help me.

**Comment by alandefreitas** [comment, marked as answer]
If the figure is in quiet mode, I believe the save function shouldn't open anything: https://alandefreitas.github.io/matplotplusplus/coding-styles/reactive-vs-quiet-figures/

In the gnuplot backend object, the save function should usually change the gnuplot terminal type temporarily and call draw. This is an implementation detail:
https://github.com/alandefreitas/matplotplusplus/blob/3fd7593f3bf98ca60fdf3db17073ea8bbe161263/source/matplot/backend/gnuplot.cpp#L148

You can try to emulate this behavior from outside, but you shouldn't have to. In any case, this might help describe the logic behind the save function.

**Reply by Darkhell101** [reply]
Thanks.  Very helpful. 
There is another problem. Chinese words are shown well in the pop window, but shown errorly when saved as png.
![第393帧](https://user-images.githubusercontent.com/37462013/154811265-bed525a1-0c49-4e94-9429-2ae1024c781d.png)

**Reply by alandefreitas** [reply]
The logic for generating the images depends on the gnuplot terminal for that file format. You could try other file formats? One of the formats is .eps, which people use a lot in academic publications. In this case, you can take the eps file and edit its internal content directly. There's also a latex output format. It creates a .tex file and the corresponding .eps file, making the output match the style of the latex template.

**Reply by Darkhell101** [reply]
I find a funny thing. When saved in pdf format, the SAVE function may do twice and generate two pages in one pdf document.
![image](https://user-images.githubusercontent.com/37462013/154830663-26f33b75-d7a5-449d-a8bb-0e1440adb02d.png)

// Quiet mode
auto f = figure(true);
auto ax = f->gca();
auto p = ax->plot(x,y);        // does not draw
p->color("red").line_width(2); // does not draw
f->show();                     // draw only once and pause console

when I try the code, compiler reminds me no PLOT method  existed.
error: ‘using element_type = class matplot::figure_type’ {aka ‘class matplot::figure_type’} has no member named ‘plot’
